src/main/java/Command.java

```diff
@@ -0,0 +1,308 @@
+     *
+     * @param splitInput A valid string array is of length 2, index 1 being the task number to
+     *     denote as done.
+     * @param keyWord The type of task.
+     * @param tasks The current list of tasks.
+     * @param storage The storage used.
+     * @return The added task message.
+     * @throws DukeException If any Duke-type exceptions are encountered.
+     */
+    public String addTaskCommand(String[] splitInput, String keyWord, TaskList tasks, Storage storage)
+            throws DukeException {
+        String[] data = processUserTaskInput(splitInput);
+        String toReturn = tasks.addTask(data, keyWord, tasks, ui);
+        storage.updateTaskFile(tasks);
+        return toReturn;
+    }
+
+    /**
+     * Obtains the keyword in the user input to determine command needed.
+     *
+     * @param input The given keyword from user input.
+     * @return The keyword if it is valid.
+     * @throws InvalidKeyWordException If the keyword not valid.
+     */
+    public String getKeyWord(String input) throws InvalidKeyWordException {
+        String[] splitInput = splitUserInput(input);
+        if (parser.isValidKeyWord(splitInput[0])) {
+            return splitInput[0];
+        } else {
+            throw new InvalidKeyWordException("\u2639 OOPS!!! I'm sorry, but I don't know what "
+                    + "that means :-(");
+        }
+    }
+
+    /**
+     * Processes the user input (for task commands only).
+     *
+     * @param array Contains strings for task type, description, date/time if applicable.
+     * @throws InvalidFormatException If the input format is wrong.
+     * @throws EmptyTaskException If input task is not tagged with description.
+     */
+    public static String[] processUserTaskInput(String[] array) throws InvalidFormatException,
+            EmptyTaskException {
+        Parser parser = new Parser();
+        if (array.length <= 1) {
+            if (array[0].equals("event")) {
+                throw new EmptyTaskException("☹ OOPS!!! The description of a event cannot "
+                        + "be empty.");
+            } else if (array[0].equals("deadline")) {
+                throw new EmptyTaskException("☹ OOPS!!! The description of a deadline cannot "
+                        + "be empty.");
+            } else if (array[0].equals("todo")) {
+                throw new EmptyTaskException("☹ OOPS!!! The description of a todo cannot "
+                        + "be empty.");
+            } else {
+                throw new InvalidFormatException("Invalid format error!");
+            }
+        } else if (array[0].equals("event")) {
+            String des = "";
+            String date = "";
+            String time = "";
+
+            boolean toBreak = false;
+            int pivotIndex = 0;
+            for (int i = 1; i < array.length; i++) {
+                if (array[i].equals("/at")) {
+                    //sets the breaking point of input
```

Comment:
Should there be a spacing after "//" for comments? I noticed the same issue elsewhere too.

Reply:
Missed that out thanks!!